smartmeter/smartmeter.ino

```diff
@@ -115,6 +127,9 @@ void setup()
     ledThread.onRun(ledTest);
     ledThread.setInterval(500);
 
+    buttonThread.onRun(readButton);
+    buttonThread.setInterval(100);
```

Comment:
This can be done with interrupt pin.

Reply:
Instead of the Thread? Wouldn't it interfere with the time measuring? I really don't know how interrupts are handled on the ESP8266.